cppcache/src/CacheXmlParser.cpp

```diff
@@ -597,7 +597,7 @@ void CacheXmlParser::startPool(const xmlChar** atts) {
         "A pool cannot be created without a name";
     throw CacheXmlException(s.c_str());
   }
-  auto factory = m_cache->getPoolManager().createFactory();
+  auto factory = std::make_shared<PoolFactory>(m_cache->getPoolManager().createFactory());
```

Comment:
Is `shared_ptr` necessary?

Reply:
Yes.  In this case `shared_ptr` is providing abstraction to push objects onto a `std::stack<std::shared_ptr<void>>` _stack.